src/status_im/contexts/wallet/wallet_connect/utils/data_store.cljs

```diff
@@ -73,3 +73,48 @@
   (-> db
       get-db-current-request-event
       get-request-params))
+
+(defn- find-networks
+  [current-request current-proposal request db]
+  (cond
+    current-request
+    (get-in current-request [:params :chainId])
+
+    current-proposal
+    (->> (get-in db [:wallet-connect/current-proposal :session-networks])
+         vec
+         (string/join ","))
+
+    request
+    (->> (or (get-in request [:params :requiredNamespaces :eip155 :chains])
+             (get-in request [:requiredNamespaces :eip155 :chains]))
+         vec
+         (string/join ","))
+
+    :else
+    []))
+
+(defn- find-url
+  [current-request current-proposal request]
+  (cond
+    current-request
+    (get-in current-request [:verifyContext :verified :origin])
+
+    current-proposal
+    (get-in current-proposal [:params :proposer :metadata :url])
+
+    request
+    (or (get-in request [:params :proposer :metadata :url])
+        (get-in request [:peer :metadata :url]))
+
+    :else
+    nil))
+
+(defn get-dapp-name-and-networks
+  [db & [request]]
+  (let [current-request  (get-in db [:wallet-connect/current-request :event])
+        current-proposal (get-in db [:wallet-connect/current-proposal :request])
```

Comment:
This is a bit confusing, what is `request` in this context? Also, we care about the "session networks" only when the user approves a proposal. For session requests we only have one network that the request is made for. They seem to be a bit different, so might be better to keep the logic for the proposal and request separated, especially since we can't really have both a `current-proposal` and `current-request` at the same time. Also will help avoid the repeated `case`s.

Reply:
Yeah make sense, will split it in 2 func (request and proposal)